packages/shared/src/components/squads/SteppedComment.tsx

```diff
@@ -1,25 +1,51 @@
-import React, { FormEventHandler, ReactElement, useContext } from 'react';
+import React, { FormEventHandler, ReactElement } from 'react';
+import { ClientError } from 'graphql-request';
 import { Modal } from '../modals/common/Modal';
 import { ModalState, SquadStateProps } from './utils';
-import { ModalPropsContext } from '../modals/common/types';
 import { SquadComment } from './Comment';
-import { SquadForm } from '../../graphql/squads';
+import { createSquad, Squad, SquadForm } from '../../graphql/squads';
+import { useToastNotification } from '../../hooks/useToastNotification';
+
+interface SteppedSquadCommentProps extends SquadStateProps {
+  onCreate?: (squad: Squad) => void;
+}
+
+const DEFAULT_ERROR = "Oops! That didn't seem to work. Let's try again!";
 
 export function SteppedSquadComment({
+  onCreate,
   onNext,
   form,
-}: SquadStateProps): ReactElement {
-  const { activeView } = useContext(ModalPropsContext);
-  if (ModalState.WriteComment !== activeView || !form.post) return null;
+}: SteppedSquadCommentProps): ReactElement {
+  const { displayToast } = useToastNotification();
   const onSubmit = (nextStep: FormEventHandler): FormEventHandler => {
-    return (e) => {
+    return async (e) => {
       e.preventDefault();
-      nextStep(e);
-      onNext({ ...form, commentary: e.target[0].value } as SquadForm);
+      const data = { ...form, commentary: e.target[0].value } as SquadForm;
+
+      if (!onCreate) {
+        nextStep(e);
+        onNext(data);
+        return;
+      }
+
+      try {
+        const newSquad = await createSquad({ ...form, ...data });
+        if (!newSquad) return;
+        onCreate(newSquad);
+        nextStep(e);
+      } catch (err) {
+        const clientError = err as ClientError;
+        const message = clientError?.response?.errors?.[0]?.message;
+        if (!message) return;
+        const error = JSON.parse(message);
+        displayToast(error?.handle ?? DEFAULT_ERROR);
```

Comment:
Should we not display the "basic" handle error in the hint field?
I think that makes more sense, the fallback could always be toast though for anything besides that.

Reply:
That is all the way back to the first step. I think we should have a more comprehensive handle checker mutation so we won't have to wait the end of the process before knowing there could be invalid formats.

I tried a handle "asdas asd" and it produced an error in the end but not on the handle check. WDYT?

Reply:
Hmm yeah it should be inline on the first step from the user perspective right?
I mean makes more sense if you preliminary know something is wrong, so maybe we should try and fix that handle check indeed.

Happy to help or debug it together if you need me 🙏

Reply:
Amazing. I will try to introduce some changes to the handle checker to see if it will be valid.

Thanks! I'll let you know if I hit a bumpy road 🚗